js/stdlib/ts/interfaces.ts

```diff
@@ -64,64 +66,79 @@ export interface Stdlib_Backend<Ty> extends Stdlib_Backend_Base<Ty> {
 export interface Stdlib_Impl_Shared {
 };
 
+// TODO: per-stddlib Provider type
+// TODO: per-stdlib env types
+// TODO: per-stdlib providerName union of string literals type
+type Provider = any
+type ProviderEnv = any // Record<string, string>
+type ProviderName = any
+
 // TODO: types
 export interface ProviderLib {
-  getProvider: any
-  setProvider: any
-  setProviderByEnv: any
-  setProviderByName: any
-  providerEnvByName: any
+  getProvider: () => Provider|Promise<Provider>
+  setProvider: (p: Provider|Promise<Provider>) => void
+  setProviderByEnv: (env: ProviderEnv) => void
+  setProviderByName: (providerName: ProviderName) => void
+  providerEnvByName: (providerName: ProviderName) => ProviderEnv
 };
 
+type FixedPoint = { sign: boolean, i: { i: BigNumber, scale: BigNumber } }
+
 // XXX
 // The shared code
 export interface Stdlib_User_Shared {
-  isBigNumber: any
-  hexToBigNumber: any
-  bigNumberToHex: any
-  uintToBytes: any
-  parseFixedPoint: any
-  parseInt: any
+  isBigNumber: (bn: unknown) => bn is BigNumber
+  hexToBigNumber: (hex: string) => BigNumber
+  bigNumberToHex: (bn: BigNumber) => string
+  uintToBytes: (bn: BigNumber) => string
+  parseFixedPoint: (fp: FixedPoint) => number
+  parseInt: (n: {sign: boolean, i: num}) => number
 };
 
 // XXX
 // The thing as composed by each connector
 export interface Stdlib_User_Base extends Stdlib_Backend_Shared_User, Stdlib_User_Shared, Arith, TypeDefs {
-  isHex: any
-  bigNumberify: any
-  stringToHex: any
-  addressEq: any
-  digest: any
+  isHex: (x: unknown) => boolean
+  bigNumberify: (n: num|string) => BigNumber
+  stringToHex: (s: string) => string
+  addressEq: (addr1: string, addr2: string) => boolean
+  digest: (t: any, a: unknown) => string
```

Comment:
`t` is a list of Types... is not included because it's hard?

Reply:
`t` is a single type (which can be an array type). I haven't tried very hard to do type params yet, but yeah it needs to be ETH ty or ALGO ty depending.